DESIGN_DOCUMENT_TEMPLATE.md

```diff
@@ -0,0 +1,139 @@
+- [Using this Document](#using-this-document)
+	- [Feature Description](#feature-description-)
+	- [Architecture Diagram](#architecture-diagram)
+		- [API Diagram](#api-diagram)
+	- [Documenting your Feature](#documenting-your-feature)
+	- [Security Reviews](#security-reviews)
+
+# Using this Document
+
+This document provides an easy-to-use template to help you get started building your own design document for OpenSearch features.
+
+To use this document you should:
+
+1. Copy and paste the contents of the document to a new file <YOUR_FEATURE>.md or to the body of a GitHub comment.
+2. Fill out the remainder of this document.
+
+**Please do not overwrite this document.**
+
+## Feature Description
+
+_Please provide a **brief** description of what feature you are introducing and what it does. This should be similar to the GitHub description._
+
+This document outlines a new feature <YOUR_FEATURE>.
+
+<YOUR_FEATURE> is...
+
+## Architecture Diagram
+
+_Some features are best explained using architecture diagrams. In particular, Mermaid diagrams are supported by GitHub and preferred. You can find examples of Mermaid diagrams in the [ARCHITECTURE.md](./ARCHITECTURE.md) file._
+
+Here is a generic graph diagram you can modify:
+
+```mermaid
+graph TD
+    subgraph Subgraph
+        subgraph Subsubgraph1
+            l1[Label 1]
+            l2[Label 2]
+        end
+        subgraph Subsubgraph2
+            l3(Label 3)
+            l4[Label 4]
+        end
+        subgraph Subsubgraph3
+           l5(Label 5)
+           l6[Label 6]
+        end
+
+        l1 -- Line 1 --> l3
+        l2 -- Line 2 --> l4
+        l3 -- Line 3 --> l5
+    end
+```
+
+
+Similarly, this is a sample sequence diagram:
+
+
+```mermaid
+sequenceDiagram
+	title Sample Sequence
+	autonumber
+	participant A as A
+	participant B as B
+	participant C as C
+	participant D as D
+	participant E as E
```

Comment:
Nit; you don't need to declare these at all so long as the links are in order, but they can be useful for friendly name vs shortname inside the diagram

Reply:
Oops I will remove them